driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -150,13 +170,73 @@ private void buildTableMetadata(KeyspaceMetadata ksm, List<Row> cfRows, Map<Stri
                         cols = Collections.<String, ColumnMetadata.Raw>emptyMap();
                     }
                 }
-                TableMetadata.build(ksm, cfRow, cols, cassandraVersion);
+
+                TableMetadata tm = TableMetadata.build(ksm, cfRow, cols, cassandraVersion);
+                TableMetadata tmPrevious = ksm.tables.get(tm.getName());
+                if (tmPrevious == null) {
+                    ksm.add(tm);
+                    triggerOnTableAdded(tm);
```

Comment:
Ah, do you mean the trigger methods in general would always be submitted on `cluster.manager.executor` inside their method definition?   That makes sense to me.  Although, how would it work when something executing inside of that executor calls these methods?  Not sure if we handle that elsewhere, but was just curious.